File: backend/dictionary.py
TSM_NAME = ['m', 't', 's']
M_DICT = {f'{TSM_NAME[0]}{i}': i for i in range(1, 10)}
T_DICT = {f'{TSM_NAME[1]}{i}': i for i in range(1, 10)}
S_DICT = {f'{TSM_NAME[2]}{i}': i for i in range(1, 10)}
WIND_DICT = {'east', 'south', 'west', 'north'}
ZFB_DICT = {'zhong', 'fa', 'bai'}
FLOWER_DICT = {'f1' : 1, 'f2' : 2, 'f3' : 3, 'f4' : 4, 'ff1' : 1, 'ff2' : 2, 'ff3' : 3, 'ff4': 4}
SEAT_DICT = {1: 'east', 2: 'south', 3: 'west', 4: 'north'}
JOKER_DICT = 'joker'

MST_DICT = M_DICT | T_DICT | S_DICT

ALL_TILES = (list(M_DICT.keys()) + list(T_DICT.keys()) + list(S_DICT.keys()) + list(WIND_DICT) + list(ZFB_DICT))
EYES_DICT = 'eyes'
SHANG_DICT = 'shang'
PONG_DICT = 'pong'

File: frontend/constants/dictionary.ts
// Taiwanese Mahjong tile dictionaries and constants

export const TSM_NAME = ["m", "t", "s"] as const;

// Generate tile dictionaries
export const M_DICT: Record<string, number> = {};
export const T_DICT: Record<string, number> = {};
export const S_DICT: Record<string, number> = {};

for (let i = 1; i <= 9; i++) {
    M_DICT[`${TSM_NAME[0]}${i}`] = i;
    T_DICT[`${TSM_NAME[1]}${i}`] = i;
    S_DICT[`${TSM_NAME[2]}${i}`] = i;
}

export const WIND_DICT = new Set(["east", "south", "west", "north"]);

export const ZFB_DICT = new Set(["zhong", "fa", "bai"]);

export const FLOWER_DICT: Record<string, number> = {
    f1: 1,
    f2: 2,
    f3: 3,
    f4: 4,
    ff1: 1,
    ff2: 2,
    ff3: 3,
    ff4: 4,
};

export const SEAT_DICT: Record<number, string> = {
    1: "east",
    2: "south",
    3: "west",
    4: "north",
};

export const JOKER_DICT = "joker";

// Combined MST dictionary
export const MST_DICT = { ...M_DICT, ...T_DICT, ...S_DICT };

// All tiles list
export const ALL_TILES = [
    ...Object.keys(M_DICT),
    ...Object.keys(T_DICT),
    ...Object.keys(S_DICT),
    ...Array.from(WIND_DICT),
    ...Array.from(ZFB_DICT),
];

// Set types
export const EYES_DICT = "eyes";
export const SHANG_DICT = "shang";
export const PONG_DICT = "pong";
export const GONG_DICT = "gong";

// Types of Hu
export const flower_hu = "花胡";
export const ligu_hu = "Ligu";
export const sixteen_bd_hu = "16不搭";
export const thirteen_waist_hu = "十三么/腰";
export const standard_hu = "普通胡";
